packages/file-adapters/lib/local-file.js

```diff
@@ -22,7 +27,11 @@ module.exports = class LocalFileAdapter {
    * Params: { stream, filename, mimetype, encoding, id }
    */
   save({ stream, filename: originalFilename, id }) {
-    const filename = `${id}-${originalFilename}`;
+    const filename = this.getFilename({ id, originalFilename });
+    if (!filename) {
+      throw new Error('LocalFileAdapter filename getter returned no or invalid value.');
```

Comment:
The only way this can fail is if the system designer passed in a bad function, so let's make this more explicit:

```
Custom function LocalFileAdapter.getFilename() returned no or invalid value.
```

Reply:
Done